app/Http/Controllers/UserController.php

```diff
@@ -125,4 +125,339 @@ public function destroy(User $user)
             return redirect()->back()->with('messageFail', 'Delete failed. Something went wrong');
         }
     }
+
+    public function testUpdateUserSuccessFeature()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $this->actingAs($admin);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
+                'email' => 'email@gmail.com',
+                'password' => '12345678',
+                'role' => '1',
+                'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+            ]
+        );
+        $params = [
+            'name' => "Name Update",
+            'email' => 'emaiupdatel@gmail.com',
+            'password' => '12345678',
+            'role' => '1',
+            'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+        ];
+        $response = $this->put(route('users.update', $user->id), $params);
+        $response->assertStatus(200)->assertJsonFragment([
+            'messageSuccess' => 'This user successfully updated',
+        ]);
+    }
+
+    public function testUnauthenticateCannotUpdateUser()
+    {
+        $admin = factory(User::class)->create(['role' => UserType::ADMIN]);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
+                'email' => 'email@gmail.com',
+                'password' => '12345678',
+                'role' => '1',
+                'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+            ]
+        );
+        $params = [
+            'name' => "Name Update",
+            'email' => 'emaiupdatel@gmail.com',
+            'password' => '12345678',
+            'role' => '1',
+            'avatar' => UploadedFile::fake()->image('avatar.jpg'),
+        ];
+        $response = $this->put(route('users.update', $user->id), $params);
+        $response->assertStatus(302);
+        $response->assertRedirect('login');
+    }
+
+    public function testUnauthorizationCannotUpdateUser()
+    {
+        $user1 = factory(User::class)->create(['role' => UserType::USER]);
+        $this->actingAs($user1);
+        $user = factory(User::class)->create(
+            [
+                'name' => "Name Create",
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Http/Controllers/UserController.php</br>**Line**: 185</br>String "Name Create" does not require double quotes; use single quotes instead